File: Inventory/Inventory.Web/Imports/ServerTypings/Inventory.PermissionKeys.ts
namespace Inventory.Inventory {
    declare namespace PermissionKeys {
        export const General = "Inventory:General";

        namespace Orders {
            export const Delete = "Inventory:Orders:Delete";
            export const Modify = "Inventory:Orders:Modify";
            export const View = "Inventory:Orders:View";
        }

        namespace Products {
            export const Delete = "Inventory:Products:Delete";
            export const Modify = "Inventory:Products:Modify";
            export const View = "Inventory:Products:View";
        }
    }
}

File: Inventory/Inventory.Web/Modules/Inventory/InventoryPermissionKeys.cs

using Serenity.ComponentModel;
using System.ComponentModel;

namespace Inventory.Inventory
{
    [NestedPermissionKeys]
    [DisplayName("Inventory")]
    public class PermissionKeys
    {
        [DisplayName("Orders")]
        public class Orders
        {
            [ImplicitPermission(General), ImplicitPermission(View)]
            public const string Delete = "Inventory:Orders:Delete";
            [Description("Create/Update"), ImplicitPermission(General), ImplicitPermission(View)]
            public const string Modify = "Inventory:Orders:Modify";
            public const string View = "Inventory:Orders:View";
        }

        [DisplayName("Products")]
        public class Products
        {
            [ImplicitPermission(General), ImplicitPermission(View)]
            public const string Delete = "Inventory:Products:Delete";
            [Description("Create/Update"), ImplicitPermission(General), ImplicitPermission(View)]
            public const string Modify = "Inventory:Products:Modify";
            public const string View = "Inventory:Products:View";
        }

        [Description("[General]")]
        public const string General = "Inventory:General";
    }
}
